demo/site/src/util/recursivelyLoadBlockData.ts

```diff
@@ -1,17 +1,21 @@
 import { type BlockLoader, type BlockLoaderDependencies, recursivelyLoadBlockData as cometRecursivelyLoadBlockData } from "@comet/site-nextjs";
 import { type AllBlockNames } from "@src/blocks.generated";
+import { loader as pageTreeIndexLoader } from "@src/common/blocks/PageTreeIndexBlock.loader";
+import { type GQLPageTreeNodeScopeInput } from "@src/graphql.generated";
 import { loader as newsDetailLoader } from "@src/news/blocks/NewsDetailBlock.loader";
 import { loader as newsListLoader } from "@src/news/blocks/NewsListBlock.loader";
 
 declare module "@comet/site-nextjs" {
     export interface BlockLoaderDependencies {
         pageTreeNodeId?: string;
+        scope?: GQLPageTreeNodeScopeInput;
```

Comment:
scope also needs to be passed for block-preview, else the block won't work in block preview. (That might be acceptable for this new block, but not for others)

While looking that this I also noticed recursivelyLoadBlockData is missing completely for FooterContentBlock. Could you please add that? (new PR)